src/query/functions/src/scalars/map.rs

```diff
@@ -245,3 +310,86 @@ pub fn register(registry: &mut FunctionRegistry) {
             },
         );
 }
+
+fn map_delete_fn(args: &[ValueRef<AnyType>], _ctx: &mut EvalContext) -> Value<AnyType> {
+    let input_length = args.iter().find_map(|arg| match arg {
+        ValueRef::Column(col) => Some(col.len()),
+        _ => None,
+    });
+
+    let mut output_map_builder = ColumnBuilder::EmptyMap { len: 0 };
```

Comment:
we can use the `return_type` to create the builder.
```rust
let mut builder = ColumnBuilder::with_capacity(&return_type, len);
```
`return_type` is same as `args_type[0]`, we can use `args_type` here if modify the `eval` as following:
```rust
eval: Box::new(move |args, ctx| {
```

Reply:
Changes implemented. Ready for your review.